bsp/src/org/jetbrains/bsp/project/resolver/BspResolverLogic.scala

```diff
@@ -288,7 +270,7 @@ private[resolver] object BspResolverLogic {
       moduleName,
       targets,
       Seq.empty, Seq.empty,
-      moduleBase,
+      moduleBase.map(_.toFile),
```

Comment:
This still won't work because you're still passing in the `id` URI, which may or may not correspond to a base directory

Reply:
Why did I touch `moduleBase`?
I pushed again reverting this part.

Reply:
Wait something is still wrong

Reply:
Using the `getBaseDirectory` instead of the `getId` for `moduleBase` won't work for Mill because both 2.12 or 2.13 for example points to the same base dir.

Reply:
SBT is not impacted for now because it gives only one `scalaVersion` not all defined in `crossScalaVersion`

Reply:
So the question is: why do you need the `basePath` to be different? Are you trying to avoid merging them into the same IntelliJ module? In that case we can consider not merging modules by path, this might be preferable anyway, because test and compile modules may have some separate configurations anyway.

Another option is reporting different base paths from Mill's side.

Reply:
You need to end up somehow with 1 module that points to `src`, another 1 pointing to `src-2.12` and another 1 pointing to `src-2.13`.
Different base path from Mill's side is not ok because that is not consistant with not cross compiled modules.

Reply:
I see. I want to investigate this more closely, but I'm still having problems importing kubernetes-client with mill-bsp, or even building it at all with vanilla mill. Maybe we can have a chat and figure that out?

Reply:
That would be great, I sent you an email.